src/fn_call.rs

```diff
@@ -15,7 +15,23 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a+'mir>: crate::MiriEvalContextExt<'a,
         dest: Option<PlaceTy<'tcx, Borrow>>,
         ret: Option<mir::BasicBlock>,
     ) -> EvalResult<'tcx, Option<&'mir mir::Mir<'tcx>>> {
+
+        // HACK: normal evaluation never sees `ReservedForIncrCompCache`, so we use it
+        // to signify a dlsym loaded thing
```

Comment:
OMG WTF.^^

Okay so at the least could we wrap this awful hack into two functions `dlsym_to_instance` and `instance_to_dlsym` (or so)?  Currently the other half of this hack is 200 lines away in the same file without any comment.

The better way, of course, would be to change `create_fn_alloc` such that it takes something like
```rust
enum MiriFunction {
  Instance(Instance),
  DlSym(String),
}
```
Then we'd need no hack.

Reply:
I know... but the compiler and const eval have zero profit from supporting this but would still have to live with the complexity.

Reply:
Having a few months perspective on this: Indeed "OMG WTF what was I thinking?" I'll teach the engine about dlsyms

Reply:
I am implementing this now. (Latest `getrandom` needs it on macOS.)